src/lwpa/socket.dox

```diff
@@ -231,23 +239,97 @@ lwpa_error_t lwpa_socket(unsigned int family, unsigned int type, lwpa_socket_t *
+ */
+typedef PLATFORM_DEFINED LwpaPollContext;
+
+/*! \brief Create a new context for waiting on events on multiple sockets.
+ *
+ *  LwpaPollContext structures contain information about an application's interest in events on a
+ *  set of sockets. The lwpa_poll_wait() function can then be used to block the calling thread until 
+ *  one such event takes place.
+ *
+ *  This function must be called on an LwpaPollContext structure before using it with any other
+ *  lwpa_poll_* function.
+ * 
+ *  \param[in,out] context Pointer to LwpaPollContext to initialize.
+ *  \return #kLwpaErrOk: Init successful.\n
+ *          #kLwpaErrInvalid: Invalid argument.\n
+ *          #kLwpaErrSys: System call failed.
+ */
+lwpa_error_t lwpa_poll_context_init(LwpaPollContext *context);
+
+/*! \brief Invalidate a previously-initialized LwpaPollContext structure.
+ * 
+ *  This must be called at shutdown or when the LwpaPollContext is no longer needed to free any
+ *  resources being used to track polling context.
+ *
+ *  \param[in] context Pointer to LwpaPollContext to deinitialize.
+ *  \return #kLwpaErrOk: Deinit successful.\n
+ *          #kLwpaErrInvalid: Invalid argument.
+ */
+lwpa_error_t lwpa_poll_context_deinit(LwpaPollContext *context);
+
+/*! \brief Add a new socket to an LwpaPollContext.
+ *
+ *  This socket will be monitored on future calls to lwpa_poll_wait() for the events specified in
+ *  the 'events' parameter. See the definitions of the LWPA_POLL_* flag values for more information.
+ *  The flag values can be 'or'ed together to monitor for multiple events. Only flag values that are
+ *  not marked 'output only' are valid for use with this function. Errors on sockets will always be
+ *  reported with the #LWPA_POLL_ERR flag.
+ *
+ *  On some systems, the underlying polling method has a limit to how many sockets can be monitored
+ *  in a single call. If there is a limit, it is defined as the positive value
+ *  #LWPA_SOCKET_MAX_POLL_SIZE. Otherwise, that constant is set to -1.
+ *
+ *  \param[in,out] context Pointer to LwpaPollContext to which to add new socket.
+ *  \param[in] socket Socket to start monitoring.
+ *  \param[in] events Events to monitor for on this socket.
+ *  \return #kLwpaErrOk: Socket added successfully.\n
+ *          #kLwpaErrInvalid: Invalid argument.\n
+ *          #kLwpaErrNoMem: #LWPA_SOCKET_MAX_POLL_SIZE is exceeded in this context.\n
+ *          #kLwpaErrSys: System call failed.
+ */
+lwpa_error_t lwpa_poll_add_socket(LwpaPollContext *context, lwpa_socket_t socket, lwpa_poll_events_t events);
+
+/*! \brief Remove a monitored socket from an LwpaPollContext.
+ *
+ *  This socket will no longer be monitored on future calls to lwpa_poll_wait().
+ *
+ *  \param[in,out] context Pointer to LwpaPollContext from which to remove socket.
+ *  \param[in] socket Socket to remove.
+ *  \return #kLwpaErrOk: Socket removed successfully.\n
```

Comment:
I'm not sure it's necessary to have this function return anything.  Is it really an error if the context was bad or the socket wasn't there?

Reply:
Yeah, good point. I'm torn on the pattern of returning status from deinit() or remove() functions. I agree that I don't see the usefulness, yet I see it in a lot of other code. If you think it's silly, I will make sure we're consistent about making all of our deinit() and remove() functions void.

Reply:
I like consistency above all. 

 If there's a case where a real system error could occur that the user should react to, then we need to return something.  Otherwise, we shouldn't. And maybe in the former case, we don't return those errors, or in the docs stress that they are warnings and not failures.

Reply:
I think if you are getting an error from a deinit function, it is probably an indication that there is some bug in your use of the functions. In this case the error code _might_ be useful for debug logging. But you'd probably be debugging anyway. I dunno.

Reply:
Alright, I'll harmonize on void deinit and remove functions.